File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    var answer = true;
    phone_book.sort();
    for(let i = 1;i<phone_book.length;i++){
        var check = phone_book[i].startsWith(phone_book[i-1]);
        if(check) {
            return false;
        }
    }
    return answer;
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <iostream>
#include <map>
using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    map<string,int> st;
    for(auto &num : phone_book){
        st[num]++;
    }
    for(int i = 0;i<phone_book.size();i++){
        string s="";
        for(int j = 0;j<phone_book[i].size()-1;j++){
            s+=phone_book[i][j];
            if(st.find(s) != st.end()){
                return false;
            }
        }
    }
    return answer;
}